astropy/timeseries/tests/test_downsample.py

```diff
@@ -0,0 +1,69 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import pytest
+import numpy as np
+from numpy.testing import assert_equal
+
+from astropy import units as u
+from astropy.time import Time
+
+from astropy.timeseries.sampled import TimeSeries
+from astropy.timeseries.downsample import simple_downsample, reduceat
+
+INPUT_TIME = Time(['2016-03-22T12:30:31', '2016-03-22T12:30:32',
+                   '2016-03-22T12:30:33', '2016-03-22T12:30:34'])
+ts = TimeSeries(time=INPUT_TIME, data=[[1, 2, 3, 4]], names=['a'])
+ts_units = TimeSeries(time=INPUT_TIME, data=[[1, 2, 3, 4] * u.count], names=['a'])
+
+
+def test_reduceat():
```

Comment:
Yeah basically only some Numpy functions have a reduceat method, so this is a way to make sure that it works for any arbitrary method. I agree it could live in astropy.utils but for now I'd rather it wasn't part of the public API until it's very stable, so I'll leave it where it is, and maybe after this PR is merged we can decided whether we want to expose it in astropy.utils.

Reply:
`reduceat` is a standard method on every numpy ufunc. It doesn't really make sense on `np.sum` since that is already equal to `np.add.reduce` (like `np.min` is equal to `np.minimum.reduce`).

But for other "reductions" like `np.median` there is indeed no analogue; anyway, bottom line is that I think it needs a bit of thought of how to expose it, so best to leave it here for the moment. But could someone open an issue so we don't forget?

Reply:
Issue opened: https://github.com/astropy/astropy/issues/8585